tests/hypervisor/test_esx.py

```diff
@@ -93,6 +93,226 @@ def test_hypervisor_id(self, virtwho, function_hypervisor, hypervisor_data, glob
+                            and result['send'] == 1
+                            and result['thread'] == 1
+                            and hypervisor_id_data not in str(result['mappings']))
+
+        function_hypervisor.update('hypervisor_id', 'uuid')
+        hypervisor_uuid = hypervisor_data['hypervisor_uuid']
+
+        # run virt-who with filter_hosts=[host_uuid] and exclude_hosts=[host_uuid]
+        function_hypervisor.update('filter_hosts', hypervisor_uuid)
+        function_hypervisor.update('exclude_hosts', hypervisor_uuid)
+        result = virtwho.run_service()
+        assert (result['error'] == 0
+                and result['send'] == 1
+                and result['thread'] == 1
+                and hypervisor_uuid not in str(result['mappings']))
+
+        # run virt-who with filter_hosts=* and exclude_hosts=[host_uuid]
+        function_hypervisor.update('filter_hosts', '*')
+        function_hypervisor.update('exclude_hosts', hypervisor_uuid)
+        result = virtwho.run_service()
+        assert (result['error'] == 0
+                and result['send'] == 1
+                and result['thread'] == 1
+                and hypervisor_uuid not in str(result['mappings']))
+
+        # run virt-who with exclude_hosts= and filter_hosts=[host_uuid]
+        function_hypervisor.update('filter_hosts', hypervisor_uuid)
+        function_hypervisor.update('exclude_hosts', '')
+        result = virtwho.run_service()
+        assert (result['error'] == 0
+                and result['send'] == 1
+                and result['thread'] == 1
+                and hypervisor_uuid in str(result['mappings']))
+
+    @pytest.mark.tier2
+    def test_wildcard_filter_and_exclude_hosts(self, virtwho, function_hypervisor, hypervisor_data):
+        """Test the filter_hosts= and exclude_hosts= option in /etc/virt-who.d/hypervisor.conf
+
+        :title: virt-who: esx: test filter_hosts and exclude_hosts function around wildcard
+        :id: fb5e45d1-9a0c-4dbe-82a1-5680bbcb702c
+        :caseimportance: High
+        :tags: tier2
+        :customerscenario: false
+        :upstream: no
+        :steps:
+            1. Set hypervisor_id=hostname.
+            2. Configure filter_hosts=hostname, run the virt-who service.
+            3. Configure filter_hosts='*', run the virt-who service.
+            4. Configure filter_hosts=wildcard, run the virt-who service.
+            5. Configure filter_hosts='*' and exclude_hosts=wildcard
+            5. set the hypervisor_id=uuid and hwuuid, run the above steps
+
+        :expectedresults:
+            2. Succeeded to run the virt-who, can find hostname in the log message
+            3. Succeeded to run the virt-who, can find hostname in the log message
+            4. Succeeded to run the virt-who, can find hostname the log message
+            5. The same as above
+
+        """
+        hypervisor_ids = ['hostname', 'uuid', 'hwuuid']
+        for hypervisor_id in hypervisor_ids:
```

Comment:
Seems missed the `hypervisor_id_data = hypervisor_data[f'hypervisor_{hypervisor_id}']`